.github/workflows/tests-coverage.yml

```diff
@@ -0,0 +1,33 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Node.js CI
+
+on:
+  push:
+    branches: ['*']
+  pull_request:
+    branches: ['*']
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+
+     
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+        
```

Comment:
How is this working? What is it caching here?
We are using yarn, but caching npm :thinking:

Reply:
It might have been caching dependencies with npm, in the [global cache directory ](https://apeople.automationanywhere.com/s/question/0D52t00001E51yGCAR/what-is-global-cache-and-what-happened-if-we-delete-the-data-from-global-cache-folder?language=en_US)
now I have changed it to yarn and added a check, so that the runner checks the users global cache before installing any dependencies from the internet